nixos/modules/services/misc/greenclip.nix

```diff
@@ -18,7 +18,11 @@ in {
       description = "greenclip daemon";
       wantedBy = [ "graphical-session.target" ];
       after    = [ "graphical-session.target" ];
-      serviceConfig.ExecStart = "${cfg.package}/bin/greenclip daemon";
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/greenclip daemon";
+        OOMScoreAdjust = "-999";
```

Comment:
Excellent point and thanks for taking the time to research it and provide such a nice summary! You are 100% right, restarting greenclip is enough in this case as it will just reload its history from disk. I'll update the PR to remove the OOM adjust.